src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingSubmissionService.java

```diff
@@ -182,7 +183,15 @@ public ProgrammingSubmission notifyPush(Long participationId, Object requestBody
         log.info("Create new programmingSubmission with commitHash: {} for participation {}", commit.getCommitHash(), participationId);
 
         programmingSubmission.setSubmitted(true);
-        programmingSubmission.setSubmissionDate(ZonedDateTime.now());
+
+        ZonedDateTime submissionDate = ZonedDateTime.now();
+        try {
+            submissionDate = versionControlService.get().getPushDate(programmingExerciseParticipation, commit.getCommitHash());
```

Comment:
I double-checked this. Gitlab seems to only provide the commit dates but not the push date. Contrary to Bitbucket which already provides the push date.

I allowed the Bitbucket service now to retrieve the date via the object if provided.
Does this change make sense?

Reply:
I think this change makes sense as it should prevent one (or more) unnecessary calls to BitBucket.
The only thing I am not too sure about is whether or not this should be done with a parameter for the getPushDate method. Personally, I have no issue with it right now, but it might be limiting in the future.

Reply:
The only other option would be to call a method to retrieve an optional date from the object and if it's null we call the getPushDate method.
But in my opinion, this overcomplicates things and my approach hides the information on how it gets retrieved properly.